### PR TITLE
Add the ci-operator config for prow into the repo

### PR DESCRIPTION
**Pull request by rendhalver**
Pushing the existing config from openshift/release so we can update the go version in-repo

### CODE REVIEW

File: .ci-operator.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: rhel-8-release-golang-1.20-openshift-4.14
```
Comment (samanthajayasinghe) [inline .ci-operator.yaml:5]: is it `1.20` or `1.21` (rhel-8-release-golang-1.21-openshift-4.16) ?

Comment (rendhalver) [reply .ci-operator.yaml:5]: This is basically what is currently defined in openshift-release:
https://github.com/openshift/release/blob/master/ci-operator/config/openshift/backplane-tools/openshift-backplane-tools-main.yaml#L2-L6

Comment (rendhalver) [reply .ci-operator.yaml:5]: As mentioned in the docs:
https://docs.ci.openshift.org/docs/architecture/ci-operator/#build-root-image

**Pull request merged**